pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -4759,10 +4759,112 @@ func TestReconcileWithPipelineResults(t *testing.T) {
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
 			runTestReconcileWithPipelineResults(t, tc.embeddedVal)
+			runTestReconcileWithFinallyResults(t, tc.embeddedVal)
 		})
 	}
 }
 
+func runTestReconcileWithFinallyResults(t *testing.T, embeddedStatus string) {
+	names.TestingSeed()
+	ps := []*v1beta1.Pipeline{parse.MustParsePipeline(t, `
+metadata:
+  name: test-pipeline
+  namespace: foo
+spec:
+  results:
+  - description: pipeline result
+    name: result
+    value: $(finally.a-task.results.aResult)
+  tasks:
+  - name: b-task
+    params:
+    - name: bParam
+      value: $(tasks.a-task.results.aResult)
+    taskRef:
+      name: b-task
+  finally:
```

Comment:
It might be good to have one finally task be a taskrun and one be a run, and each one has some results that get propagated to pipeline results (just to make sure we're testing both custom tasks and tasks)

Reply:
1. Have 2 tasks in `finally`, one for custom and one for tasks

Reply:
done